### PR TITLE
Prune and upgrade npm package dependencies

### PR DESCRIPTION
**Pull request by peterbom**
Updates dependencies to the latest versions, prior to enabling Dependabot to do this for us.

Change summary:
- Unused dependencies have been removed
- `@types` packages have been removed where packages publish their own types
- Node version is updated (to prevent issues where more recent packages require later node versions)
- TypeScript (and associated build/lint tools) are updated to latest versions
- Some missing dependencies have been added (`glob` and `request`)
- Additions to `tsconfig.json` to prevent TypeScript analysing the `node_modules` directory and enable the recommended `esModuleInterop` setting)
- Dependency updates and fixes for breaking changes

This [builds, runs and passes automated tests](https://github.com/peterbom/vscode-kubernetes-tools/actions/runs/8515519844), but it's probably worth doing some further manual testing.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -17,7 +17,9 @@
         "noImplicitAny": true,
         "strictFunctionTypes": true,
         "strictNullChecks": true,
-        "noImplicitThis": true
+        "noImplicitThis": true,
+        "esModuleInterop": true,
```
Comment (peterbom) [inline tsconfig.json:21]: Setting [esModuleInterop](https://www.typescriptlang.org/tsconfig#esModuleInterop) to true is [recommended](https://github.com/microsoft/TypeScript/issues/41139) and required for [compatibility](https://github.com/kubernetes-client/javascript/issues/751) with later versions of the K8s client library.

File: tsconfig.json [hunk 2]
```diff
@@ -17,7 +17,9 @@
...
+        "skipLibCheck": true
```
Comment (peterbom) [inline tsconfig.json:22]: Without setting this, we end up with type errors in the [K8s client](https://github.com/kubernetes-client/javascript/issues/864), the [glob types](https://github.com/isaacs/node-glob/issues/529), and [shelljs](https://github.com/shelljs/shelljs/issues/1148).

**Review by Tatsinnit — APPROVED** [review]
Thanks for this. Looks good to me, I gave a quick test for the vsix shared, and seems alright and things work normal. ☕️🙏

For anyone keen here is the `vsix` to test if any wider community folks what to chip in. Thanks

[vscode-kubernetes-tools-1.3.16-20240402-01.vsix.zip](https://github.com/vscode-kubernetes-tools/vscode-kubernetes-tools/files/14984616/vscode-kubernetes-tools-1.3.16-20240402-01.vsix.zip)

**Review by Moazzem-Hossain-pixel — COMMENTED** [review]
Thanks for your cooperation, I will glad if you can take the necessary action for resolving the issues.

**Review by Moazzem-Hossain-pixel — COMMENTED** [review]
Thanks for your contributions.

**Pull request merged**